docs/architecture/adr-009-proof-size-analysis.md

```diff
@@ -0,0 +1,155 @@
+# ADR 009: New Non-Interactive Default Rules for Reduced Padding
+
+## Changelog
+
+- 14.11.2022: Initial Draft
+
+## Context
+
+ Is it worth it to change non-interactive default rules as a follow-up of [ADR 008](./adr-008-square-size-independent-message-commitments.md)  to decrease the padding? The downside is that the message inclusion proof size will not be as efficient in big square sizes.
+
+ **Note: This analysis implies the implementation of #1004. If the tree over the subtree roots is not a Namespace Merkle Tree then the analysis of both methods has the same proof size.**
+
+- Inter-message padding can be reduced if we can change the non-interactive default rules from this:
+
+    > - Messages that span multiple rows must begin at the start of a row (this can occur if a message is longer than k shares or if the block producer decides to start a message partway through a row and it cannot fit).
+    > - Messages begin at a location aligned with the largest power of 2 that is not larger than the message length or k.
+
+    To this: **Messages start at an index that is a multiple of its `msgMinSquareSize`.**
+
+    As an example, we have this diagram. Message 1 is three shares long and is followed by message 2, which is 11 shares long, so the `msgMinSquareSize` of the second message is equal to four. Therefore we have a padding of 5 shares shown in light blue. Furthermore, with the new non-interactive default rule set, a message of size 11 can start in this block at index zero and index three because they are multiples of four. Therefore, we save four shares of padding while retaining the same commitment.
+
+    ![Padding Savings](./assets/padding-savings.png)  
```

Comment:
[optional] I noticed a few small styling things in this section and rather than suggesting changes on individual lines, I suggested a revised section. The suggestions are based on:
- reduce one layer of indentation
- visually format the current and proposed non-interactive default rules so that they are easier to compare
- link to issue #1004 
- describe upside and downside in the same paragraph
- reword the padding savings explanation and separate the current vs proposed into distinct paragraphs

```suggestion
## Context

[ADR 008](./adr-008-square-size-independent-message-commitments.md) makes it possible for the current non-interactive default rules to be modified slightly:

Current

> - Messages that span multiple rows must begin at the start of a row (this can occur if a message is longer than k shares or if the block producer decides to start a message partway through a row and it cannot fit).
> - Messages begin at a location aligned with the largest power of 2 that is not larger than the message length or k.

Proposed

> - Messages start at an index that is a multiple of its `msgMinSquareSize`.

The upside of this proposal is that it reduces the inter-message padding. The downside is that a message inclusion proof will not be as efficient for large square sizes so the proof will be larger.

> **Note**
> This analysis assumes the implementation of [celestia-app#1004](https://github.com/celestiaorg/celestia-app/issues/1004). If the tree over the subtree roots is not a Namespace Merkle Tree then both methods have the same proof size.**

As an example, take the diagram below. Message 1 is 3 shares long and message 2 is 11 shares long.

With the current non-interactive default rules, message 2 must start at a location aligned with the largest power of 2 that is not larger than 11 (message length) or 8 (square size). Therefore, message 2 must start at a location aligned with 8 which is index 16 in the example below. This arragenemnt results in 5 shares of padding.

With the proposed non-interactive default rules, message 2 must start at an index that is a multiple of `msgMinSquareSize`. A message of 11 shares can fit in a square size of 4 (since 4 * 4 = 16 available shares which is < 11) so `msgMinSquareSize` is 4. Therefore, message 2 can start at index 12. This arrangement results in 1 share of padding.

![Padding Savings](./assets/padding-savings.png)
```